src/net_processing.cpp

```diff
@@ -2714,10 +2743,21 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         // set nodes not capable of serving the complete blockchain history as "limited nodes"
         pfrom.m_limited_node = (!(nServices & NODE_NETWORK) && (nServices & NODE_NETWORK_LIMITED));
 
-        if (peer->m_tx_relay != nullptr) {
+        // We only initialize the m_tx_relay data structure if:
+        // - this isn't an outbound block-relay-only connection; and
+        // - fRelay=true or we're offering NODE_BLOOM to this peer
+        //   (NODE_BLOOM means that the peer may turn on tx relay later)
+        if (!pfrom.IsBlockOnlyConn() &&
+            (fRelay || (pfrom.GetLocalServices() & NODE_BLOOM))) {
+            auto* tx_relay = [&peer]() {
+                LOCK(peer->m_tx_relay_mutex);
+                Assume(!peer->m_tx_relay);
```

Comment:
#21596 :smiling_face_with_tear: 

```
net_processing.cpp:2754:31: error: reading variable 'm_tx_relay' requires holding mutex 'peer->m_tx_relay_mutex' [-Werror,-Wthread-safety-analysis]
                Assume(!peer->m_tx_relay);
                              ^
1 error generated.

Reply:
Yeah, I'll remove the lambda.

Reply:
I think the issue isn't with the lambda, but with `Assume`/`Assert`.

Reply:
Thanks. Removed.

Reply:
Just for reference, an alternative to removing would be to temporarily asssign the value to a bool and then `Assume(name_of_bool)`.

Reply:
Another reference: If #24714 is merged, you may be able to re-add the Assume as it was.

Reply:
I'll leave this for now, and add the `Assume()` back if 24714 is merged. I'm 100% sure that this code path can only get hit once for each peer, so the `Assume()` was only there to catch weird regressions.

Reply:
> I'll leave this for now, and add the Assume() back if 24714 is merged.

Now done.